docs/contributing/coding-guidelines/ros-nodes/class-design.md

```diff
@@ -1,5 +1,67 @@
 # Class design
 
+We'll use the `autoware_gnss_poser` package as an example.
+
+## Namespaces
+
+```cpp
+namespace autoware::gnss_poser
+{
+...
+} // namespace autoware::gnss_poser
+```
+
+- Everything should be under `autoware::gnss_poser` namespace.
+- Closing braces should contain a comment with the namespace name.
+
+## Classes
+
+### Nodes
+
+#### `gnss_poser.hpp`
+
+```cpp
+class GNSSPoserNode : public rclcpp::Node
+{
+  public:
+    explicit GNSSPoserNode(const rclcpp::NodeOptions & node_options);
+  ...
+}
+```
+
+#### `gnss_poser.cpp`
+
+```cpp
+GNSSPoserNode::GNSSPoserNode(const rclcpp::NodeOptions & node_options)
+: Node("gnss_poser_node", node_options)
+{
+  ...
+}
+```
+
+- The class name should be in `CamelCase`.
+- Node classes should inherit from `rclcpp::Node`.
+- The constructor should be explicit.
+- The constructor should take `rclcpp::NodeOptions` as an argument.
+- Default node name:
+  - should not have `autoware_` prefix.
+  - should have `_node` suffix.
```

Comment:
I couldn't get why the default name should have a "_node" suffix.
Redundant suffixes may lose readability when we list up nodes by `ros2 node list`, and I don't come up with a case of confusion that names of nodes and others being mixed.

Reply:
Updated the docs with https://github.com/autowarefoundation/autoware-documentation/pull/547/commits/77b3039a7302445d48aee511f7b9eb0a9850dd4b :+1: